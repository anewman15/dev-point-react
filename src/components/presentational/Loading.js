import React from 'react';

const Loading = () => (
  <div>
    <img src={`${process.env.PUBLIC_URL}/loading.gif`} alt="loading" />
  </div>
);

export default Loading;
