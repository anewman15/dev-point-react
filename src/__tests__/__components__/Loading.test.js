import { unmountComponentAtNode } from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Loading from '../../components/presentational/Loading';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('Loading', () => {
  it('renders the Loading component to the dom', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Loading />
        </MemoryRouter>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
