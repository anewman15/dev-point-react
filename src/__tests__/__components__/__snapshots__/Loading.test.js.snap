// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders the Loading component to the dom 1`] = `
<div>
  <img
    alt="loading"
    src="/loading.gif"
  />
</div>
`;
